lib/src/mock_document_snapshot.dart

```diff
@@ -20,14 +20,19 @@ class MockDocumentSnapshot extends Mock implements DocumentSnapshot {
   String get id => _id;
 
   @override
-  dynamic get(dynamic key) => _document![key];
+  dynamic get(dynamic key) {
+    if (_isComposedKey(key)) {
+      return getComposedKeyValue(key);
+    }
+    return _document![key];
+  }
 
   @override
-  Map<String, dynamic>? data() {
+  Map<String, dynamic> data() {
```

Comment:
Actually, `DocumentSnapshot.data()` is still nullable, so we shouldn't change its signature here. See [DocumentSnapshot's documentation](https://pub.dev/documentation/cloud_firestore/latest/cloud_firestore/DocumentSnapshot-class.html#instance-methods).

It's `QueryDocumentSnapshot.data()` that was made to be non-nullable. For this issue, @lyledean1 kindly fixed the issue in #153.